### PR TITLE
Removed unnecessary code from codable clases

### PR DESCRIPTION
**Pull request by mcousillas6**
#### Description:

* Removed unnecessary initializers and encode/decode functions, since Codable already synthetizes them for us

Reference: https://www.swiftbysundell.com/basics/codable

### CODE REVIEW

File: ios-base/Networking/APIService.swift [hunk 1]
```diff
@@ -74,6 +74,7 @@ open class BaseApiService<T>: APIService where T: TargetType {
   open var jsonDecoder: JSONDecoder {
     let decoder = JSONDecoder()
     decoder.keyDecodingStrategy = .convertFromSnakeCase
+    decoder.dateDecodingStrategy = .millisecondsSince1970
```
Comment (germanStabile) [inline ios-base/Networking/APIService.swift]: 👏

File: ios-base/Networking/APIService.swift [hunk 2]
```diff
@@ -74,6 +74,7 @@ open class BaseApiService<T>: APIService where T: TargetType {
   open var jsonDecoder: JSONDecoder {
     let decoder = JSONDecoder()
     decoder.keyDecodingStrategy = .convertFromSnakeCase
```
Comment (glm4) [inline ios-base/Networking/APIService.swift]: will this take care of `Session. accessToken`?

Comment (mcousillas6) [reply ios-base/Networking/APIService.swift]: Nope, that's why session still has `CodingKeys`

Comment (glm4) [reply ios-base/Networking/APIService.swift]: perfect!

**Pull request merged**